src/main/scala/com/gu/rcspollerlambda/services/MetadataService.scala

```diff
@@ -24,6 +29,13 @@ object MetadataService extends Logging {
     }
   }
 
+  // A grid ID should be 40 characters long and valid hex characters
+  def isGridId(potentialId: String): Boolean = {
+    // include upper and lowercase to be safe
+    val validChars = "0123456789abcdefABCDEF".toList
```

Comment:
could this be done with regex? or is this more readable would you say?

Reply:
This should be faster and is arguably more readable. It should however be a set rather than a list to be optimal. Not that we need to be in this context.